docs/tutorials/k8s.md

```diff
@@ -0,0 +1,145 @@
+# Sifchain - Kubernetes (k8s) Tutorial
+
+#### Demo Videos
+
+
+#### Dependencies:
+
+- Clone the repository (`git clone git@github.com:Sifchain/sifnode.git`)
+- [Ruby 2.7.x](https://www.ruby-lang.org/en/documentation/installation)
+- [Golang](https://golang.org/doc/install)
+- [AWS CLI Tool](https://aws.amazon.com/cli/)
+- [kubectl](https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html)
+- [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+_This tutorial assumes that you have at least a basic understanding of setting up AWS and configuring your access keys accordingly, so that you may interact with AWS via the CLI Tool._
+
+#### What is Kubernetes? (k8s)
+
+Kubernetes is an open-source container-orchestration system for automating application deployment, scaling, and management.
+
+## Scaffold and deploy a new cluster
+
+1. Switch to the root of the sifchain project.
+
+2. Scaffold a new cluster:
+
+```
+rake "cluster:scaffold[<chainID>,<provider>]"
+```
+
+where:
+
+|Param|Description|
+|-----|----------|
+|<chainID>|The Chain ID of the network (e.g.: merry-go-round).|
+|<provider>|The cloud provider to use (currently only AWS is supported).|
+
+3. Once complete, you'll notice that several Terraform files/folders have been setup inside of the `.live` directory. We recommend you leave the defaults as-is, but for those that have experience with Terraform, feel free to adjust the configuration as you see fit.
+
+4. Deploy the cluster to AWS:
+
+```
+rake "cluster:deploy[<chainID>,<provider>]"
+```
+
+5. Once complete, you should see your cluster on your AWS account. You can also check using `kubectl`:
+
+```
+kubectl get pods --all-namespaces
+```
+
+## Deploy a new cluster
+
+1. Generate a new mnemonic key for your node. This key is what your node will use to eventually sign transactions/blocks on the network.
+
+```
+rake "keys:generate:mnemonic"
+```
+
+2. Import your newly generated key:
+
+```
+rake "keys:import[<moniker>]"
+```
+
+where:
+
+|Param|Description|
+|-----|----------|
+|<moniker>|The moniker or name of your node as you want it to appear on the network.|
+
+3. Check that it's been imported accordingly:
+
+```
+sifnodecli keys show <moniker> --keyring-backend file 
+```
+
+4. Deploy a new node to your cluster and connect to an existing network:
+
+```
+rake "cluster:sifnode:deploy:peer[<chainID>,<provider>,<namespace>,<image>,<image tag>,<moniker>,<mnemonic>,<peer address>,<genesis URL>]"
```

Comment:
id also add concrete examples too. As some of these are optional etc And it would be good to just be able to copy and paste as much as possible too.